### PR TITLE
Bug fix by removing test playbook to not be installed along with content pack

### PR DESCRIPTION
**Pull request by moishce**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29587

## Description
When inside the ״TestPlaybook״ directory, there was only the "NonCircleTests" directory
So the tests in "NonCircleTests" would not be deleted

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py [hunk 1]
```diff
@@ -642,6 +642,12 @@ def remove_unwanted_files(self, delete_test_playbooks=True):
         task_status = True
 
         try:
+            pack_dirs = [f for f in os.listdir(self._pack_path)]
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/marketplace_services.py:645]:
```suggestion
            pack_dirs = [os.listdir(self._pack_path)]
```

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -642,6 +642,12 @@ def remove_unwanted_files(self, delete_test_playbooks=True):
...
+            for directory in Pack.EXCLUDE_DIRECTORIES:
+                if delete_test_playbooks and directory in pack_dirs:
+                    shutil.rmtree(f'{self._pack_path}/{directory}')
+                    print(f"Deleted pack {directory} directory for {self._pack_name} pack")
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/marketplace_services.py:649]:
```suggestion
                    print(f"Deleted {directory} directory from {self._pack_name} pack")
```

**Review by ShahafBenYakir — COMMENTED** [review]
1. Let's add a unit test.
2. Can I see a faulty build before this fix and another that succeeds?
3. Add changelog.

**Comment by ShahafBenYakir** [comment]
@moishce Please upload here two links for two builds that demonstrate before and after the fix

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 3]
```diff
@@ -284,6 +285,22 @@ def test_is_feed(self, yaml_context, yaml_type, is_actually_feed):
         dummy_pack.is_feed_pack(yaml_context, yaml_type)
         assert dummy_pack.is_feed == is_actually_feed
 
+    def test_remove_unwanted_files(self):
+        """
+           Given:
+               - Pack name & path.
+           When:
+               - Before uploading a pack to the marketplace, delete unwanted files.
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/Tests/marketplace_services_test.py:293]:
```suggestion
               - Preparing packs before uploading to marketplace.
```

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 4]
```diff
@@ -284,6 +285,22 @@ def test_is_feed(self, yaml_context, yaml_type, is_actually_feed):
...
+           Then:
+               - Assert the TestPlaybooks die was deleted.
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/Tests/marketplace_services_test.py:295]:
```suggestion
               - Assert `TestPlaybooks` directory was deleted from pack.
```

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 5]
```diff
@@ -284,6 +285,22 @@ def test_is_feed(self, yaml_context, yaml_type, is_actually_feed):
...
+       """
+        os.mkdir('Tests/Marketplace/Tests/test_data/pack_to_test')
+        os.mkdir('Tests/Marketplace/Tests/test_data/pack_to_test/TestPlaybooks')
+        test_pack = Pack(pack_name="pack_to_test", pack_path='Tests/Marketplace/Tests/test_data/pack_to_test')
+        test_pack.remove_unwanted_files()
+        assert not os.path.isdir('Tests/Marketplace/Tests/test_data/pack_to_test/TestPlaybooks')
+        shutil.rmtree('Tests/Marketplace/Tests/test_data/pack_to_test')
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/Tests/marketplace_services_test.py:302]: Let's add here another folder like "integrations" and check that it was not deleted and also another folder inside the TestPlaybooks (like we had that deprecated folder)

**Comment by moishce** [comment]
[link to the build before the fix](https://app.circleci.com/pipelines/github/demisto/content/35832/workflows/38d2cb6d-498d-448b-b9ac-a4925e9a76aa/jobs/156290/parallel-runs/0/steps/0-113)
[link to the build after the fix](https://app.circleci.com/pipelines/github/demisto/content/35833/workflows/ff5f11d9-8865-4e1b-951e-6746f6ef74b2/jobs/156304/parallel-runs/0/steps/0-113)

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 6]
```diff
@@ -290,15 +290,20 @@ def test_remove_unwanted_files(self):
            Given:
                - Pack name & path.
            When:
-               - Before uploading a pack to the marketplace, delete unwanted files.
+               - Preparing packs before uploading to marketplace.
            Then:
-               - Assert the TestPlaybooks die was deleted.
+               - Assert `TestPlaybooks` directory was deleted from pack.
+               - Assert `TestPlaybooks` directory was not deleted from pack.
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/Tests/marketplace_services_test.py:296]:
```suggestion
               - Assert `Integrations ` directory was not deleted from pack.
```

**Review by ShahafBenYakir — APPROVED** [review]
Just remove  the unnecessary changes you made to test this and we are good to go

**Pull request merged**